src/pyatsimagebuilder/utils.py

```diff
@@ -70,6 +85,42 @@ def git_clone(url, path, commit_id=None, rm_git=False):
     return hexsha
 
 
+def clone_with_credentials(url, path, credentials):
+    from urllib.parse import urlparse
+    url = urlparse(url)
+    url = url._replace(netloc='{}:{}@{}'.format(
+        credentials['username'], credentials['password'], url.netloc))
```

Comment:
credentials only guaranteed to have 'username' and 'password' if schema is updated to reflect this.